### PR TITLE
General doctor cleanup and refactoring

### PR DESCRIPTION
**Pull request by mlissner**
This lands a handful of general maintenance stuff, but not a whole lot else:

1. Better docs for users and devs
2. Code is blackified according to our usual black config, with a pyproject.toml file added.
3. Old URL aliases are removed.
4. Unnecessary URLs are removed.
5. Docker container names are tweaked to less likely to collide with CourtListener.
6. Removed duplicated dependencies in requirements.txt and creates requirements.dev.txt with a few dev dependencies.

### CODE REVIEW

**Comment by mlissner** [comment]
Forgot to mention. Fixes: #149

**Comment by mlissner** [comment]
@flooie, can you please take a look at this when you get a sec?

**Comment by flooie** [comment]
Apologies

**Comment by mlissner** [comment]
Any thoughts. :) ?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,9 +14,7 @@ jobs:
       uses: actions/setup-python@v1
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Create docker network
-      run: docker network create -d bridge --attachable cl_net_overlay
     - name: Build Image
       run: docker-compose -f docker-compose.dev.yml up --build -d
     - name: Run tests
```
Comment (flooie) [inline .github/workflows/tests.yml:19]: IM surprised its not needed for tests but great

File: doctor/views.py [hunk 2]
```diff
@@ -123,7 +108,7 @@ def extract_doc_content(request) -> Union[JsonResponse, HttpResponse]:
 
     # Get page count if you can
     page_count = get_page_count(fp, extension)
-    os.remove(form.cleaned_data["fp"])
+    cleanup_form(form)
```
Comment (flooie) [inline doctor/views.py:111]: this makes me wonder...

File: requirements.txt [hunk 3]
```diff
@@ -1,13 +1,9 @@
 Django==3.2.6
 gunicorn==20.1.0
 PyPDF2[crypto]
-Pillow
-requests
-pdf2image
 certifi
 chardet>=3.0.4
 eyed3
-ipython
```
Comment (flooie) [inline requirements.txt:10]: good drop of dependencies. but im shocked we can drop pillow

**Review by flooie — COMMENTED** [review]
Nothing too radical.  here this looks good to me as tests did pass.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mlissner) [reply .github/workflows/tests.yml:19]: Yeah, it uses a simpler kind of network if you don't create one manually.

File: doctor/views.py (hunk 2 above)
Comment (mlissner) [reply doctor/views.py:111]: `cleanup_form` does the same thing as `os.remove...`, so no change in functionality here, I think.

File: requirements.txt (hunk 3 above)
Comment (mlissner) [reply requirements.txt:10]: I think these were all listed twice.

Comment (flooie) [reply requirements.txt:10]: ah

**Pull request merged**